nvflare/edge/ios/NVFlareMobile/NVFlareMobile/Sources/Trainers/ETTrainer/ETDataset.cpp

```diff
@@ -0,0 +1,112 @@
+#include "ETDataset.hpp"
+#include <executorch/extension/tensor/tensor.h>
+#include <sstream>
+
+using namespace executorch::extension;
+
+// Constants
+const int IMAGE_SIZE = 32 * 32 * 3;  // 3072 bytes per image
+const int NUM_IMAGES = 10000;        // Each batch has 10,000 images
```

Comment:
pre-splitted data file with fixed number of images (num_images and datafile itself needs to be well-aligned I guess)? shall we have a more self-contained way for feeding dataset file?

Reply:
fixed